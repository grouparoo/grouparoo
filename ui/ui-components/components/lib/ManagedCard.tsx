import { Card, CardProps, Col, Row } from "react-bootstrap";
import Loader from "../Loader";
import SeparatedItems from "./SeparatedItems";

interface Props {
  title: string;
  actions?: React.ReactNode[];
  disabled?: boolean;
  border?: CardProps["border"];
  loading?: boolean;
}

const ManagedCard: React.FC<Props> = ({
  title,
  actions,
  disabled,
  border,
  children,
  loading,
}) => {
  const slugTitle = title.toLowerCase().replace(/ /, "-");
  return (
    <Card className={disabled ? "text-muted" : undefined} border={border}>
      <Card.Header>
        <Row>
          <Col>
            <h5 id={slugTitle} className="my-1">
              {title}{" "}
              {loading && (
                <Loader style={{ verticalAlign: "inherit" }} size="sm" />
              )}
            </h5>
          </Col>
          {actions && (
            <Col className="text-right" md="7" lg="6" sm="12">
              <SeparatedItems items={actions} />
            </Col>
          )}
        </Row>
      </Card.Header>
      {children}
    </Card>
  );
};

export default ManagedCard;
