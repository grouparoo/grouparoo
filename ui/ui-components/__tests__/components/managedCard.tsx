import { render } from "@testing-library/react";
import ManagedCard from "../../components/lib/ManagedCard";

describe("managedCard", () => {
  it("will render a spinner when in the loading state", async () => {
    let card = render(<ManagedCard title="test" loading={false}></ManagedCard>);
    expect(card.container).toMatchSnapshot("noLoading");
    card = render(<ManagedCard title="test" loading={true}></ManagedCard>);
    expect(card.container).toMatchSnapshot("loading");
  });
});
