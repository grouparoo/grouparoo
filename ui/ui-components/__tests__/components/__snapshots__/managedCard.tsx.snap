// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`managedCard will render a spinner when in the loading state: loading 1`] = `
<div>
  <div
    class="card"
  >
    <div
      class="card-header"
    >
      <div
        class="row"
      >
        <div
          class="col"
        >
          <h5
            class="my-1"
            id="test"
          >
            test
             
            <div
              class="spinner-border spinner-border-sm"
              role="status"
              style="vertical-align: inherit;"
            />
          </h5>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`managedCard will render a spinner when in the loading state: noLoading 1`] = `
<div>
  <div
    class="card"
  >
    <div
      class="card-header"
    >
      <div
        class="row"
      >
        <div
          class="col"
        >
          <h5
            class="my-1"
            id="test"
          >
            test
             
          </h5>
        </div>
      </div>
    </div>
  </div>
</div>
`;
